ada/views.py

```diff
@@ -0,0 +1,38 @@
+import re
+from urllib.parse import urljoin, urlparse
+
+from django.http import Http404, HttpRequest, HttpResponse
+from django.shortcuts import HttpResponseRedirect, render
+
+from .models import RedirectUrl, RedirectUrlsEntry
+
+
+def clickActivity(request: HttpRequest, pk: int, whatsappid: str) -> HttpResponse:
+    try:
+        redirect_url = RedirectUrl.objects.get(id=pk)
+    except (ValueError, RedirectUrl.DoesNotExist):
+        raise Http404()
+    else:
+        store_url_entry = RedirectUrlsEntry(symptom_check_url=redirect_url)
+        store_url_entry.save()
+        url = f"{redirect_url.symptom_check_url}"
+        qs = f"?whatsappid={whatsappid}"
+        parsed_url = urlparse(qs)
+        destination_url = urljoin(url, parsed_url.geturl())
+        return HttpResponseRedirect(destination_url)
+
+
+def default_page(request: HttpRequest, pk: int) -> HttpResponse:
+    regex = re.compile("[@_!#$%^&*()<>?/\|}{~:]")
```

Comment:
These are not all possible special characters.. What if I have a \`, or a 😀 . 

I think it's better just to properly encode and forward on any special characters in the URL, rather than returning an error here.